src/components/DetailsDrillDownAlert/DetailsAlertsMetrics.js

```diff
@@ -0,0 +1,188 @@
+import PropTypes from 'prop-types'
+import { useDispatch, useSelector } from 'react-redux'
+import { isEmpty } from 'lodash'
+
+import ApplicationMetricCard from '../DetailsMetrics/ApplicationMetricCard'
+import DatePicker from '../../common/DatePicker/DatePicker'
+import NoData from '../../common/NoData/NoData'
+import NoMetricData from '../DetailsMetrics/NoMetricData'
+
+import { REQUEST_CANCELED } from '../../constants'
+import detailsActions from '../../actions/details'
+import modelEndpointsActions from '../../actions/modelEndpoints'
+import { groupMetricByApplication } from '../../elements/MetricsSelector/metricsSelector.util'
+
+import {
+  datePickerPastOptions,
+  PAST_24_HOUR_DATE_OPTION,
+  TIME_FRAME_LIMITS
+} from '../../utils/datePicker.util'
+
+const DetailsAlertsMetrics = ({ selectedItem }) => {
+  const [metrics, setMetrics] = useState([])
+  const [requestErrorMessage, setRequestErrorMessage] = useState('')
+  const metricsContainerRef = useRef(null)
+  const metricsValuesAbortController = useRef(new AbortController())
+  const prevSelectedEndPointNameRef = useRef('')
+  const [metricOptionsAreLoaded, setMetricOptionsAreLoaded] = useState(false)
+  const detailsStore = useSelector(store => store.detailsStore)
+  const dispatch = useDispatch()
+
+  const generatedMetrics = useMemo(() => {
+    return groupMetricByApplication(metrics, true)
+  }, [metrics])
+
+  const handleChangeDates = useCallback(
+    (dates, isPredefined, selectedOptionId) => {
+      const generatedDates = [...dates]
+
+      if (generatedDates.length === 1) {
+        generatedDates.push(new Date())
+      }
+
+      dispatch(
+        detailsActions.setDetailsDates({
+          value: generatedDates,
+          selectedOptionId,
+          isPredefined
+        })
+      )
+    },
+    [dispatch]
+  )
+
+  useEffect(() => {
+    const past24hoursOption = datePickerPastOptions.find(
+      option => option.id === PAST_24_HOUR_DATE_OPTION
+    )
+
+    handleChangeDates(past24hoursOption.handler(), true, PAST_24_HOUR_DATE_OPTION)
+  }, [handleChangeDates])
+
+  useEffect(() => {
+    dispatch(
+      modelEndpointsActions.fetchModelEndpointMetrics(selectedItem.project, selectedItem.uid)
+    ).then(() => setMetricOptionsAreLoaded(true))
+  }, [dispatch, selectedItem.project, selectedItem.uid])
+
+  const fetchData = useCallback(
+    (params, projectName, uid) => {
+      metricsValuesAbortController.current = new AbortController()
+      return dispatch(
+        modelEndpointsActions.fetchModelEndpointMetricsValues(
+          projectName,
+          uid,
+          params,
+          metricsValuesAbortController.current,
+          setRequestErrorMessage
+        )
+      ).then(metrics => {
+        if (metrics) setMetrics(metrics)
+      })
+    },
+    [dispatch, setMetrics, metricsValuesAbortController]
+  )
+
+  useEffect(() => {
+    if (selectedItem.uid !== prevSelectedEndPointNameRef.current) {
```

Comment:
can we move all this logic to a fncton?

Reply:
I've added the fetchMetrics function.
Let me know if this is what you intended.